test/Test-EndToEnd.js

```diff
@@ -216,6 +216,184 @@ describe('EndToEnd', () => {
         });
     });
 
+    describe('FileStore with relativeLocation', () => {
+        let file_id;
+        let deferred_file_id;
+        before(() => {
+            server = new Server();
+            server.datastore = new FileStore({
+                path: STORE_PATH,
+                // configure the store to return relative path in Location Header
+                relativeLocation: true
+            });
+            agent = request.agent(server.listen());
+        });
+
+        after((done) => {
+            // Remove the files directory
+            exec(`rm -r ${FILES_DIRECTORY}`, (err) => {
+                if (err) {
+                    return done(err);
+                }
+
+                // clear the config
+                server.datastore.configstore.clear();
+                return done();
+            });
+        });
+
+        describe('HEAD', () => {
+            it('should 404 file ids that dont exist', (done) => {
+                agent.head(`${STORE_PATH}/${file_id}`)
+                .set('Tus-Resumable', TUS_RESUMABLE)
+                .expect(404)
+                .expect('Tus-Resumable', TUS_RESUMABLE)
+                .end(done);
+            });
+        });
+
+        describe('POST', () => {
+            it('should create a file that will be deleted', (done) => {
+                agent.post(STORE_PATH)
+                .set('Tus-Resumable', TUS_RESUMABLE)
+                .set('Upload-Defer-Length', 1)
+                .set('Tus-Resumable', TUS_RESUMABLE)
+                .expect(201)
+                .end((err, res) => {
+                    assert.equal('location' in res.headers, true);
+                    assert.equal(res.headers['tus-resumable'], TUS_RESUMABLE);
+                    // Save the id for subsequent tests
+                    file_to_delete = res.headers.location.split('/').pop();
+                    done();
+                });
+            });
+
+            it('should create a file and respond with its _relative_ location', (done) => {
```

Comment:
this is really the only new test you need, regressions would be caught in the existing tests

Reply:
ok, i removed the unnecessary tests